plugin/evm/blockgascost/cost.go

```diff
@@ -0,0 +1,57 @@
+// (c) 2025, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+// blockgascost implements the block gas cost logic
+package blockgascost
+
+import (
+	"math"
+
+	safemath "github.com/ava-labs/avalanchego/utils/math"
+	"github.com/ava-labs/subnet-evm/commontype"
+)
+
+// BlockGasCost calculates the required block gas cost.
+//
+// cost = parentCost + step * (TargetBlockRate - timeElapsed)
+//
+// The returned cost is clamped to [MinBlockGasCost, MaxBlockGasCost].
+func BlockGasCost(
```

Comment:
seems this method has a different type than coreth, but the code is very similar to ap4 package with additional parameters.

probably later we can consider merging these code paths?

Reply:
yes, the main thing is we need to support dynamic configurations here so we need to pass fee configs.